### PR TITLE
Improve CI for Android artifacts

### PR DESCRIPTION
**Pull request by rpavlik**
Updates our actions, actually pushes stuff to the release, etc.

### CODE REVIEW

File: .github/workflows/android-snapshot.yml [hunk 1]
```diff
@@ -12,12 +12,14 @@ jobs:
...
       - name: Install ninja-build tool
         uses: seanmiddleditch/gha-setup-ninja@v3
+
       - name: set up JDK 11
-        uses: actions/setup-java@v2
+        uses: actions/setup-java@v3.6.0
         with:
           java-version: "11"
-          distribution: "adopt"
+          distribution: "temurin"
```
Comment (utzcoz) [inline .github/workflows/android-snapshot.yml:20]: `zulu` also a great recommendation distribution for JDK.

File: .github/workflows/android-snapshot.yml [hunk 2]
```diff
@@ -12,12 +12,14 @@ jobs:
       - uses: actions/checkout@v2
       - name: Install ninja-build tool
         uses: seanmiddleditch/gha-setup-ninja@v3
+
       - name: set up JDK 11
-        uses: actions/setup-java@v2
+        uses: actions/setup-java@v3.6.0
```
Comment (utzcoz) [inline .github/workflows/android-snapshot.yml:17]: What about using `setup-java@3`? Current v3.6.0 is specific style and `v3` is a reference style, and I remember GitHub will use latest specific version when running.

Comment (utzcoz) [reply .github/workflows/android-snapshot.yml:17]: If using specific version for GitHub Actions, it's recommended to enable dependabot bot of GitHub to auto-update GitHub Actions version automatically with sent PRs. But I afraid that dependabot can't pass OpenXR's contributor license signing requirement.

Comment (rpavlik) [reply .github/workflows/android-snapshot.yml:17]: Oh interesting, I thought it only took repo tags. I'll look into that.

And I'll ask the Khronos web person about dependabot, thanks

**Pull request merged**